NuRadioReco/detector/RNO_G/db_mongo_read.py

```diff
@@ -825,25 +826,34 @@ def get_channel_signal_chain(self, channel_signal_id, measurement_name=None, ver
         # and do not describe own components on their own ("channel" and "breakout")
         filtered_component_dict = {}
         additional_information = {}
+        weight_dict = {}
 
         for key, ele in component_dict.items():
-            if re.search("(channel|breakout)", key) is None:
+            if re.search("(channel|breakout|weight)", key) is None:
                 filtered_component_dict[key] = ele
+            elif re.search("weight", key) is not None:
+                weight_dict[key.replace("_weight", "")] = ele
             else:
                 additional_information[key] = ele
             
         components_data = {}
         for component, component_id in filtered_component_dict.items():
             # Add the additional informatio which were filtered out above to the correct components
-            supp_info = {k.replace(component + "_", ""): component_dict[k] for k in component_dict 
-                         if component in k and component != k}
+            supp_info = {k.replace(component + "_", ""): additional_information[k] for k in additional_information 
+                         if re.search(component, k)}
 
-            component_data = self.get_channel_signal_chain_component_data(
-                component, component_id, supp_info, primary_time=self.__database_time, verbose=verbose)
+            if re.search("golden", component, re.IGNORECASE):
```

Comment:
Can you explain that?

Reply:
Yes, I am currently working on implementing the weights in the signal chain. In order to do so, we need to save the Golden components (that were used to make the IGLU, DRAB or Fiber measurements) in the signal chain. As a result, we need to take care of these components in the query part.

Reply:
Why?